clients/src/main/resources/common/message/ListGroupsRequest.json

```diff
@@ -20,8 +20,14 @@
   // Version 1 and 2 are the same as version 0.
   //
   // Version 3 is the first flexible version.
-  "validVersions": "0-3",
+  //
+  // Version 4 adds the States flexible field (KIP-518).
+  "validVersions": "0-4",
   "flexibleVersions": "3+",
   "fields": [
+    { "name": "States", "type": "[]string", "versions": "4+", "tag": 0, "taggedVersions": "4+",
```

Comment:
Sorry I missed this from the discussion, but why are we bumping the version if we are only adding tagged fields? Is it so that we can detect whether the capability is supported? If so, then I wonder why we don't make this a regular field.

Reply:
Yes the version bump is necessary to detect if this field was supported.

As we're bumping the version and as you said in https://github.com/apache/kafka/pull/8238#discussion_r426869755 the overhead of the extra field on this API is not a concern, it's probably simpler to use a regular field.